homeassistant/components/konnected/config_flow.py

```diff
@@ -0,0 +1,561 @@
+"""Config flow for konnected.io integration."""
+import asyncio
+from collections import OrderedDict
+import copy
+import logging
+
+import voluptuous as vol
+from voluptuous.humanize import humanize_error
+
+from homeassistant import config_entries
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_DOOR,
+    DEVICE_CLASSES as BIN_SENS_TYPES,
+    DEVICE_CLASSES_SCHEMA,
+)
+from homeassistant.const import (
+    CONF_BINARY_SENSORS,
+    CONF_HOST,
+    CONF_ID,
+    CONF_NAME,
+    CONF_PORT,
+    CONF_SENSORS,
+    CONF_SWITCHES,
+    CONF_TYPE,
+    CONF_ZONE,
+)
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+
+from .const import (
+    CONF_ACTIVATION,
+    CONF_BLINK,
+    CONF_DISCOVERY,
+    CONF_INVERSE,
+    CONF_MOMENTARY,
+    CONF_PAUSE,
+    CONF_POLL_INTERVAL,
+    CONF_REPEAT,
+    DOMAIN,
+    STATE_HIGH,
+    STATE_LOW,
+    ZONES,
+)
+from .errors import CannotConnect
+from .panel import get_status
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_KONN_UPNP_MODEL_NAME = "model_name"  # standard upnp is modelName
+
+KONN_MANUFACTURER = "konnected.io"
+KONN_MODEL = "Konnected"
+KONN_MODEL_PRO = "Konnected Pro"
+KONN_PANEL_MODEL_NAMES = [KONN_MODEL, KONN_MODEL_PRO]
+
+DATA_SCHEMA_MANUAL = OrderedDict()
+DATA_SCHEMA_MANUAL[vol.Required("host")] = str
+DATA_SCHEMA_MANUAL[vol.Required("port")] = int
+
+
+DATA_SCHEMA_KONN_MODEL = OrderedDict()
+DATA_SCHEMA_KONN_MODEL[vol.Required("1", default="Disabled")] = vol.In(
+    ["Disabled", "Binary Sensor", "Digital Sensor", "Switchable Output"]
+)
+DATA_SCHEMA_KONN_MODEL[vol.Required("2", default="Disabled")] = vol.In(
+    ["Disabled", "Binary Sensor", "Digital Sensor", "Switchable Output"]
+)
+DATA_SCHEMA_KONN_MODEL[vol.Required("3", default="Disabled")] = vol.In(
+    ["Disabled", "Binary Sensor", "Digital Sensor", "Switchable Output"]
+)
+DATA_SCHEMA_KONN_MODEL[vol.Required("4", default="Disabled")] = vol.In(
+    ["Disabled", "Binary Sensor", "Digital Sensor", "Switchable Output"]
+)
+DATA_SCHEMA_KONN_MODEL[vol.Required("5", default="Disabled")] = vol.In(
+    ["Disabled", "Binary Sensor", "Digital Sensor", "Switchable Output"]
+)
+DATA_SCHEMA_KONN_MODEL[vol.Required("6", default="Disabled")] = vol.In(
+    ["Disabled", "Binary Sensor", "Digital Sensor", "Switchable Output"]
```

Comment:
is there a way to clean up this repetition?

Reply:
Yep - will create a few sub schemas for the different IO types.